.gitignore

```diff
@@ -0,0 +1,229 @@
+
+# Created by https://www.gitignore.io/api/macos,python,intellij+all
+
+### Intellij+all ###
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and WebStorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff
+.idea/**/workspace.xml
```

Comment:
You can just do `.idea/*`. No need to add individual directory/files.

Reply:
@song-hao, are we expecting any dotfiles to be in this repo?
If not, we could just put in something like `.*` to ignore all the dotfiles you listed here